lab_3_generate_by_ngrams/main.py

```diff
@@ -270,6 +415,33 @@ def run(self, seq_len: int, prompt: str) -> Optional[str]:
         In case of corrupt input arguments or methods used return None,
         None is returned
         """
+        if not isinstance(seq_len, int) or not isinstance(prompt, str) or len(prompt) == 0:
+            return None
+
+        encoded_text_tuple = self._text_processor.encode(prompt)
+        if encoded_text_tuple is None:
+            return None
+        # context_len = self._model.get_n_gram_size() - 1
+        encoded_text_list = list(encoded_text_tuple)
+
+        for i in range(seq_len):
+            possible_tokens = self._model.generate_next_token(encoded_text_tuple)
+            if possible_tokens is None:
+                return prompt + '.'
```

Comment:
I believe that dot is not an identifier of any token

Reply:
I have all tests passed only when there is a dot at the end of decoded text

Reply:
fixed